### PR TITLE
Addons docs configurable

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- add flag to Helm-Broker, which enables docs and docs topics creation
- refactor integration tests

**Related issue(s)**
Related: https://github.com/kyma-project/kyma/issues/5211

### CODE REVIEW

File: test/integration/integration_test.go [hunk 1]
```diff
@@ -347,29 +289,42 @@ func TestDocsTopic(t *testing.T) {
...
+	})
+
+	t.Run("cluster", func(t *testing.T) {
+		suite.assertNoServicesInCatalogEndpoint("cluster")
+
+		// when
+		suite.createClusterAddonsConfiguration("addon1", []string{accTestRepo}, sourceHTTP)
+
+		// then
+		suite.waitForClusterAddonsConfigurationPhase("addon1", v1alpha1.AddonsConfigurationReady)
```
Comment (polskikiel) [inline test/integration/integration_test.go]: please add `assertClusterDocsTopicListIsEmpty()` check for `namespaced` and `cluster` case

File: internal/controller/setup.go [hunk 2]
```diff
@@ -18,6 +18,11 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
+type docsFacade interface {
```
Comment (polskikiel) [inline internal/controller/setup.go]: minor: you may move that interface definition to `ext.go` file

File: test/integration/integration_test.go (hunk 1 above)
Comment (piotrmiskiewicz) [reply test/integration/integration_test.go]: CRDs are not installed so I cannot test it

File: internal/controller/setup.go (hunk 2 above)
Comment (piotrmiskiewicz) [reply internal/controller/setup.go]: moved

**Pull request merged**